src/xunit.analyzers/X1000/TheoryDataTypeArgumentsShouldBeSerializable.cs

```diff
@@ -0,0 +1,567 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Xunit.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class TheoryDataTypeArgumentsShouldBeSerializable : XunitDiagnosticAnalyzer
+{
+	const string IsNotSerializable = "is not serializable";
+	const string MightNotBeSerializable = "might not be serializable";
+
+	public TheoryDataTypeArgumentsShouldBeSerializable() :
+		base(Descriptors.X1044_TheoryDataTypeArgumentsShouldBeSerializable)
+	{ }
+
+	public override void AnalyzeCompilation(
+		CompilationStartAnalysisContext context,
+		XunitContext xunitContext)
+	{
+		Guard.ArgumentNotNull(context);
+		Guard.ArgumentNotNull(xunitContext);
+
```

Comment:
Can you explain this? I don't understand the reasoning here.

Reply:
The GAC only exists in .NET Framework. It doesn't exist for .NET Core/.NET.

Reply:
I realize now I really put this comment in the wrong place and with nowhere near enough words. 😝 We can skip "Enum in the GAC" analysis when the context is analyzing a .NET Framework targeting project.

Reply:
Ok, that makes more sense! 😂

Reply:
Could you clarify how to determine whether the context is analyzing a project targeting .NET Framework?

Reply:
I've never done it myself, so I don't know. I assume it would be part of the context that's given, but I'd have to poke around it in a debugger to figure out how (and whether it's even possible), which is how I do 100% of my Roslyn work, sadly. 😂

Reply:
Since you don't know, we can leave it off for now and I'll try to add it later.

Reply:
> which is how I do 100% of my Roslyn work, sadly.

😂

I poked around the context aggregate, and briefly searched online, but didn't find anything.

I'll spend a little time later investigating how feasible it would be to check whether Roslyn has source-level access to an enum definition, like you suggested.

If it's not, and if we can't figure out how to check the target framework, then I suppose the enum serializability problem is intractable after all. In which case, the analyzer should presumably continue ignoring all enum types? Since otherwise, every single enum type would produce the "might not be serializable" diagnostic, which I imagine would be more bothersome than useful.

Reply:
> I'll spend a little time later investigating how feasible it would be to check whether Roslyn has source-level access to an enum definition, like you suggested.

I was able to dig this out in the debugger. 😁 Looks like you can look for `ISourceAssemblySymbol` on the type's `ContainingAssembly`. Here's one where the type is local:

```
> type.ContainingAssembly is ISourceAssemblySymbol
true
```

And here's one where it isn't:

```
> context.Compilation.GetTypeByMetadataName("System.ConsoleColor").ContainingAssembly is ISourceAssemblySymbol
false
```

The actual concrete types were helpfully named `SourceAssemblySymbol` and `NonSourceAssemblySymbol` 😆 

Still looking for access to target framework...

Reply:
I think the more I think about it, we're only circling around to something vaguely successful for enums, so we should just ignore enums entirely. The GAC requirement isn't reliably testable, and honestly most people these days are probably targeting .NET (and not .NET Framework), so let's just let the .NET Framework people fend for themselves, which is exactly where they would've been before this analyzer.

Reply:
> The actual concrete types were helpfully named `SourceAssemblySymbol` and `NonSourceAssemblySymbol` 😆

Thanks, good to know!

> I think the more I think about it, we're only circling around to something vaguely successful for enums, so we should just ignore enums entirely. The GAC requirement isn't reliably testable, and honestly most people these days are probably targeting .NET (and not .NET Framework), so let's just let the .NET Framework people fend for themselves, which is exactly where they would've been before this analyzer.

Alright, sounds good.